linode/vpcips/datasource_test.go

```diff
@@ -0,0 +1,71 @@
+//go:build integration || vpc_ips
+
+package vpcips_test
+
+import (
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/linode/terraform-provider-linode/v2/linode/acceptance"
+	"github.com/linode/terraform-provider-linode/v2/linode/vpcips/tmpl"
+)
+
+func TestAccDataSourceVPCIPs_basic(t *testing.T) {
+	t.Parallel()
+
+	resourceName_all_ips := "data.linode_vpc_ips.foobar"
+	resourceName_vpc_ips := "data.linode_vpc_ips.barfoo"
+	resourceName_filtered_ips := "data.linode_vpc_ips.foobar_filter"
+
+	vpcLabel := acctest.RandomWithPrefix("tf-test")
+	testRegion, err := acceptance.GetRandomRegionWithCaps([]string{"VPCs"})
+	if err != nil {
+		log.Fatal(fmt.Errorf("Error getting region: %s", err))
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acceptance.PreCheck(t) },
+		ProtoV5ProviderFactories: acceptance.ProtoV5ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: tmpl.DataBasic(t, vpcLabel, testRegion, "10.0.0.0/24", "10.0.1.0/24"),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName_all_ips, "vpc_ips.#", "2"),
+					resource.TestCheckResourceAttr(resourceName_vpc_ips, "vpc_ips.#", "1"),
```

Comment:
Assertions like these may fail when there are other VPC instances live in the same account, which is very likely for our shared testing account because multiple tests may run concurrently. I would suggest adding filter to every test case to ensure only the VPC created by this test workflow are selected by the data source in the template files.

Reply:
Good point, I didn't consider this. I updated the first test case to filter on the ids of the two newly created linodes. For the second test, I did not add a filter because the datasource is given the id of the newly created vpc, so it should only ever return the IP Addresses relevant to the test. Let me know if you think I should add a filter anyway.

Reply:
That looks great! Thanks for addressing it!